app/src/main/kotlin/com/intive/picover/validators/TextValidator.kt

```diff
@@ -0,0 +1,37 @@
+package com.intive.picover.validators
+
+class TextValidator private constructor(
+	private val text: String,
+	private val allowEmpty: Boolean,
+	private val allowBlank: Boolean,
+	private val maxLength: Int,
+) {
+
+	fun isValidated(): Boolean {
+		if (!allowEmpty && text.isEmpty()) {
+			return false
+		}
+
+		if (!allowBlank && text.isBlank()) {
+			return false
+		}
+
+		if (text.length > maxLength) {
+			return false
+		}
+
+		return true
+	}
+
+	class Builder(private val text: String) {
+		private var allowEmpty: Boolean = true
+		private var allowBlank: Boolean = true
+		private var maxLength: Int = 0
```

Comment:
I wonder if wouldn't be better to make these properties nullable instead of default values, and then in function `isValidated()` check only things which are not null

Reply:
Done 😄